Modules/ITS/src/TrendingTaskITSFhr.cxx

```diff
@@ -0,0 +1,327 @@
+// Copyright CERN and copyright holders of ALICE O2. This software is
+// distributed under the terms of the GNU General Public License v3 (GPL
+// Version 3), copied verbatim in the file "COPYING".
+//
+// See http://alice-o2.web.cern.ch/license for full licensing information.
+//
+// In applying this license CERN does not waive the privileges and immunities
+// granted to it by virtue of its status as an Intergovernmental Organization
+// or submit itself to any jurisdiction.
+
+///
+/// \file    TrendingTaskITSFhr.cxx
+/// \author  Ivan Ravasenga on the structure from Piotr Konopka
+///
+
+#include "ITS/TrendingTaskITSFhr.h"
+#include "../../../Framework/src/RootClassFactory.h"
+#include "QualityControl/DatabaseInterface.h"
+#include "QualityControl/MonitorObject.h"
+#include "QualityControl/QcInfoLogger.h"
+#include "QualityControl/Reductor.h"
+#include "../../Common/include/Common/TH2XlineReductor.h"
+#include <TCanvas.h>
+#include <TH1.h>
+#include <map>
+#include <string>
+
+using namespace o2::quality_control;
+using namespace o2::quality_control::core;
+using namespace o2::quality_control::postprocessing;
+using namespace o2::quality_control_modules::common;
+
+void TrendingTaskITSFhr::configure(std::string name,
+                                   const boost::property_tree::ptree& config)
+{
+  mConfig = TrendingTaskConfigITS(name, config);
+}
+
+void TrendingTaskITSFhr::initialize(Trigger,
+                                    framework::ServiceRegistry& services)
+{
+  // Preparing data structure of TTree
+  mTrend = std::make_unique<TTree>(); // todo: retrieve last TTree, so we
+                                      // continue trending. maybe do it
+                                      // optionally?
+  mTrend->SetName(PostProcessingInterface::getName().c_str());
+  //mTrend->Branch("meta", &mMetaData, "runNumber/I");
+  mTrend->Branch("runNumber", &mMetaData.runNumber);
+  mTrend->Branch("ntreeentries", &ntreeentries);
+  mTrend->Branch("time", &mTime);
+
+  for (const auto& source : mConfig.dataSources) {
+    std::unique_ptr<Reductor> reductor(root_class_factory::create<Reductor>(
+      source.moduleName, source.reductorName));
+    if (source.reductorName.find("TH2Xline") != std::string::npos) {
+      TH2XlineReductor::mystat* mystc = (TH2XlineReductor::mystat*)reductor->getBranchAddress();
+      mTrend->Branch(Form("%s_mean", source.name.c_str()), &(mystc->mean));
+      mTrend->Branch(Form("%s_stddev", source.name.c_str()), &(mystc->stddev));
+      mTrend->Branch(Form("%s_entries", source.name.c_str()), &(mystc->entries));
+      mTrend->Branch(Form("%s_occupancy", source.name.c_str()), &(mystc->mean_scaled));
```

Comment:
So, just out of curiosity, what will be stored with branches specified in this manner? Each entry in a branch will contain the full growing vector or perhaps its current last element?

Reply:
it's the current last element but now that I'm thinking I never clean the vectors. Uhm, let me check this (I'm using just a single run for my tests and I didn't see issues but there could be something strange).